### ISSUE TITLE
`experimental_useObject` isLoading doesn't turns false after error

### ISSUE THREAD

**Issue by sahanatvessel** [vercel/ai#2876]
### Description

```
const {submit, isLoading, object, error} = experimental_useObject({
    api: '/api/chat',
    schema: z.array(baseJobSchema),
    onFinish({object}) {
      if (object != null && object.length > 0) {
        setJobs(prev => [...object.map(el => el), ...prev])
      }

      conversionInProgress.current = false
    },
    onError: () => {
      toast.error("You've been rate limited, please try again later!")
      conversionInProgress.current = false
    }
  })


```

In the api/chat throw an error to see the isLoading is still true

### Code example

```
const {submit, isLoading, object, error} = experimental_useObject({
    api: '/api/chat',
    schema: z.array(baseJobSchema),
    onFinish({object}) {
      if (object != null && object.length > 0) {
        setJobs(prev => [...object.map(el => el), ...prev])
      }

      conversionInProgress.current = false
    },
    onError: () => {
      toast.error("You've been rate limited, please try again later!")
      conversionInProgress.current = false
    }
  })


```

If there is an error isLoading should revert to false?

### Additional context

[N/A]

**Comment by lgrammel** [comment]
Fix: https://github.com/vercel/ai/pull/2877

**Comment by lgrammel** [comment]
Fixed in `ai@3.3.26`

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This is a concrete failure in existing hook state behavior on the error path, and maintainers confirmed it required a fix.

### Evidence
- The reporter states that `experimental_useObject` leaves `isLoading` set to `true` when the API throws an error, instead of resetting it to `false`.
- A maintainer linked PR #2877 as a fix and confirmed the issue was fixed in `ai@3.3.26`.
